OpenRA.Mods.Common/Activities/Attack.cs

```diff
@@ -69,26 +71,78 @@ public override Activity Tick(Actor self)
 			if (attackStatus.HasFlag(AttackStatus.NeedsToMove))
 				return moveActivity;
 
+			if (attackStatus.HasFlag(AttackStatus.FrozenSwitch))
+			{
+				// When shooting a non-frozen target which became frozen,
+				// this makes sure that the target line will remain even after
+				// the underlying actor is destroyed
+				self.SetTargetLine(frozenSwitchActivity.Target, Color.Red);
```

Comment:
No, the opposite, there was a case where the unit continued shoooting, but the red line was gone once the real actor died. This makes sure that the red line remains in its place even when the actor dies.